### PR TITLE
sdk/framework: add TypeSignedDurationSecond FieldType

### PR DESCRIPTION
**Pull request by mgaffney**
Adds the `TypeSignedDurationSecond` `FieldType` which accepts positive and negative durations. The existing `TypeDurationSecond` `FieldType` does not accept negative durations.

Will close #6979 in favor of this PR.

### CODE REVIEW

File: sdk/framework/field_data_test.go [hunk 1]
```diff
@@ -202,6 +202,105 @@ func TestFieldDataGet(t *testing.T) {
 			0,
 		},
```
Comment (kalafut) [inline sdk/framework/field_data_test.go]: Since 0 is the edge condition, I'd add tests against 0 for both the old and new type.

**Comment by kalafut** [comment]
Minor additional test case suggestion but otherwise LGTM.

File: sdk/framework/field_data_test.go (hunk 1 above)
Comment (mgaffney) [reply sdk/framework/field_data_test.go]: Good idea
